source/common/upstream/outlier_detection_impl.cc

```diff
@@ -456,100 +473,62 @@ void DetectorImpl::runCallbacks(HostSharedPtr host) {
+        detector.successRateAverage());
+    event.mutable_eject_success_rate_event()
+        ->mutable_cluster_success_rate_ejection_threshold()
+        ->set_value(detector.successRateEjectionThreshold());
+    event.mutable_eject_success_rate_event()->mutable_host_success_rate()->set_value(
+        host->outlierDetector().successRate());
+  } else {
+    event.mutable_eject_consecutive_event();
   }
+
+  const auto json = MessageUtil::getJsonStringFromMessage(event, /* pretty_print */ false,
+                                                          /* always_print_primitive_fields */ true);
+  file_->write(fmt::format("{}\n", json));
 }
 
 void EventLoggerImpl::logUneject(HostDescriptionConstSharedPtr host) {
-  // TODO(mattklein123): Log friendly host name (e.g., instance ID or DNS name).
-  // clang-format off
-  static const std::string json =
-    std::string("{{") +
-    "\"time\": \"{}\", " +
-    "\"secs_since_last_action\": \"{}\", " +
-    "\"cluster\": \"{}\", " +
-    "\"upstream_url\": \"{}\", " +
-    "\"action\": \"uneject\", " +
-    "\"num_ejections\": {}" +
-    "}}\n";
-  // clang-format on
-  SystemTime now = time_source_.systemTime();
-  MonotonicTime monotonic_now = time_source_.monotonicTime();
-  file_->write(fmt::format(
-      json, AccessLogDateTimeFormatter::fromTime(now),
-      secsSinceLastAction(host->outlierDetector().lastEjectionTime(), monotonic_now),
-      host->cluster().name(), host->address()->asString(), host->outlierDetector().numEjections()));
-}
+  envoy::data::cluster::v2alpha::OutlierDetectionEvent event;
 
-std::string EventLoggerImpl::typeToString(EjectionType type) {
-  switch (type) {
-  case EjectionType::Consecutive5xx:
-    return "5xx";
-  case EjectionType::ConsecutiveGatewayFailure:
-    return "GatewayFailure";
-  case EjectionType::SuccessRate:
-    return "SuccessRate";
-  }
+  absl::optional<MonotonicTime> time = host->outlierDetector().lastEjectionTime();
+  setCommonEventParams(event, host, time);
 
-  NOT_REACHED_GCOVR_EXCL_LINE;
+  event.set_action(envoy::data::cluster::v2alpha::Action::UNEJECT);
+
+  const auto json = MessageUtil::getJsonStringFromMessage(event, /* pretty_print */ false,
+                                                          /* always_print_primitive_fields */ true);
+  file_->write(fmt::format("{}\n", json));
 }
 
-int EventLoggerImpl::secsSinceLastAction(const absl::optional<MonotonicTime>& lastActionTime,
-                                         MonotonicTime now) {
-  if (lastActionTime) {
-    return std::chrono::duration_cast<std::chrono::seconds>(now - lastActionTime.value()).count();
+void EventLoggerImpl::setCommonEventParams(
+    envoy::data::cluster::v2alpha::OutlierDetectionEvent& event, HostDescriptionConstSharedPtr host,
+    absl::optional<MonotonicTime>& time) {
+  MonotonicTime monotonic_now = time_source_.monotonicTime();
+  if (time) {
+    int secsFromLastAction =
```

Comment:
nit: `std::chrono::seconds secs_from_last_action`, and then use `count()` in the set call below.